sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetUtils.scala

```diff
@@ -208,6 +208,9 @@ object ParquetUtils {
     case st: StructType =>
       sqlConf.parquetVectorizedReaderNestedColumnEnabled &&
         st.fields.forall(f => isBatchReadSupported(sqlConf, f.dataType))
+    case udt: UserDefinedType[_] =>
+      sqlConf.parquetVectorizedReaderNestedColumnEnabled &&
```

Comment:
why do we need to check `sqlConf.parquetVectorizedReaderNestedColumnEnabled`? if `udt.sqlType` is `AtomicType` then this should return true.

Reply:
Yes, I can update that.